soc/arm/nordic_nrf/Kconfig

```diff
@@ -16,6 +16,12 @@ config SOC_FAMILY
 source "soc/arm/nordic_nrf/Kconfig.peripherals"
 source "soc/arm/nordic_nrf/*/Kconfig.soc"
 
+config SOC_SECURE_NRF
+	bool
+	default y if !TRUSTED_EXECUTION_NONSECURE
```

Comment:
If this option is to stay, I'd suggest to define it as:
```
	def_bool !TRUSTED_EXECUTION_NONSECURE || BUILD_WITH_TFM
```

Reply:
You mean like this?
```
config NRF_SOC_SECURE_SUPPORTED
    def_bool !TRUSTED_EXECUTION_NONSECURE
```

I'm not familiar with how def_bool works, is it just a shorthand for a bool with a default value?

What I want in downstream is something like this:
```
config SPM
    select NRF_SOC_SECURE_SUPPORTED
```